patches/tModLoader/Terraria/Item.cs.patch

```diff
@@ -916,10 +916,11 @@
  		bool flag = true;
  		while (flag) {
  			flag = false;
+-			if (rolledPrefix == -1 && unifiedRandom.Next(4) == 0)
+-				rolledPrefix = 0;
 +
 +			if (applyPrefixOverride is true) { } else
- 			if (rolledPrefix == -1 && unifiedRandom.Next(4) == 0)
--				rolledPrefix = 0;
++			if (prefixWeWant == -1 && unifiedRandom.Next(4) == 0)
 +				return true; //rolledPrefix = 0;
```

Comment:
Comment out the vanilla code with `/* and */` to make patch clearer.

Add a comment with `//` explaining why we check `prefixWeWant` instead of `rolledPrefix`

No need for the weird `else` on same line formatting, since we're not trying to keep the existing vanilla lines anymore. You can just combine the `applyPrefixOverride is not true &&` into the condition

Reply:
The reason for checking prefixWeWant instead of rolledPrefix is that the unexplained change to returning breaks the part below that forces it try again if no prefix is rolled with prefixWeWant equal to -2;
I don't believe I can sufficiently explain this change without knowing why there was a change that needed to be fixed in the first place, given it seems to originate from [1.4.4.2: Reimplement prefix hooks](https://github.com/tModLoader/tModLoader/commit/2204125d59b16374806b9663f27fbe7a3214632d), was the original change simply intended to improve performance?